telegram/_chatboost.py

```diff
@@ -0,0 +1,408 @@
+from telegram._utils.argumentparsing import parse_sequence_arg
+from telegram._utils.datetime import extract_tzinfo_from_defaults, from_timestamp
+from telegram._utils.types import JSONDict
+
+if TYPE_CHECKING:
+    from telegram import Bot
+
+
+class ChatBoostSource(TelegramObject):
+    """
+    Base class for Telegram ChatBoostSource objects. It can be one of:
+
+    :class:`telegram.ChatBoostSourcePremium`
+    :class:`telegram.ChatBoostSourceGiftCode`
+    :class:`telegram.ChatBoostSourceGiveaway`
+
+    Objects of this class are comparable in terms of equality. Two objects of this class are
+    considered equal, if their :attr:`source` is equal.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        source (:obj:`str`): The source of the chat boost. Can be one of:
+            :attr:`~telegram.ChatBoostSource.PREMIUM`, :attr:`~telegram.ChatBoostSource.GIFT_CODE`,
+            or :attr:`~telegram.ChatBoostSource.GIVEAWAY`.
+
+    Attributes:
+        source (:obj:`str`): The source of the chat boost. Can be one of:
+            :attr:`~telegram.ChatBoostSource.PREMIUM`, :attr:`~telegram.ChatBoostSource.GIFT_CODE`,
+            or :attr:`~telegram.ChatBoostSource.GIVEAWAY`.
+    """
+
+    __slots__ = ("source",)
+
+    PREMIUM: Final[str] = "premium"  # TODO: Make it a reference to constants
+    """:const:`telegram.constants.ChatBoostSources.PREMIUM`"""
+    GIFT_CODE: Final[str] = "gift_code"
+    """:const:`telegram.constants.ChatBoostSources.GIFT_CODE`"""
+    GIVEAWAY: Final[str] = "giveaway"
+    """:const:`telegram.constants.ChatBoostSources.GIVEAWAY`"""
+
+    def __init__(self, source: str, *, api_kwargs: Optional[JSONDict] = None):
+        super().__init__(api_kwargs=api_kwargs)
+
+        # Required by all subclasses:
+        self.source: str = source
+
+        self._id_attrs = (self.source,)
+        self._freeze()
+
+    @classmethod
+    def de_json(cls, data: Optional[JSONDict], bot: "Bot") -> Optional["ChatBoostSource"]:
+        """See :meth:`telegram.TelegramObject.de_json`."""
+        data = cls._parse_data(data)
+
+        if not data:
+            return None
+
+        _class_mapping: Dict[str, Type[ChatBoostSource]] = {
+            cls.PREMIUM: ChatBoostSourcePremium,
+            cls.GIFT_CODE: ChatBoostSourceGiftCode,
+            cls.GIVEAWAY: ChatBoostSourceGiveaway,
+        }
+
+        if cls is ChatBoostSource and data.get("source") in _class_mapping:
+            return _class_mapping[data.pop("source")].de_json(data=data, bot=bot)
+
+        if "user" in data:
+            data["user"] = User.de_json(data.get("user"), bot)
+
+        return super().de_json(data=data, bot=bot)
+
+
+class ChatBoostSourcePremium(ChatBoostSource):
+    """
+    The boost was obtained by subscribing to Telegram Premium or by gifting a Telegram Premium
+    subscription to another user.
+
+    .. versionadded:: NEXT.VERSION
+
+    Args:
+        source (:obj:`str`): The source of the chat boost. Always
```

Comment:
no source argument needed. attribute can be removed as well, b/c documented in base class :) some for the others

Reply:
yep, realized that and many changes waiting to be pushed once I figure out the tests